core/unit_test/TestViewCopy.hpp

```diff
@@ -50,126 +50,244 @@
+      ASSERT_TRUE(run_check(hs_b, 5));
+    }
+    {
+      Kokkos::deep_copy(s_a, hs_a);
+      ASSERT_TRUE(run_check(s_a, 6));
+    }
+  }
 
-}  // namespace
+  // Contiguous copies
+  {
+    Kokkos::deep_copy(dev, a, 2);
+    ASSERT_TRUE(run_check(a, 2));
+  }
+  {
+    Kokkos::deep_copy(dev, b, 3);
+    ASSERT_TRUE(run_check(b, 3));
+  }
+  {
+    Kokkos::deep_copy(dev, h_a, 4);
+    ASSERT_TRUE(run_check(h_a, 4));
+  }
+  {
+    Kokkos::deep_copy(dev, a, b);
+    ASSERT_TRUE(run_check(a, 3));
+  }
+  {
+    Kokkos::deep_copy(dev, h_b, h_a);
+    ASSERT_TRUE(run_check(h_b, 4));
+  }
+  {
+    Kokkos::deep_copy(dev, h_a, a);
+    ASSERT_TRUE(run_check(h_a, 3));
+  }
+  {
+    Kokkos::deep_copy(dev, b, h_b);
+    ASSERT_TRUE(run_check(b, 4));
+  }
+  // Non contiguous copes
+  {
+    Kokkos::deep_copy(dev, s_a, 5);
+    ASSERT_TRUE(run_check(s_a, 5));
+  }
+  {
+    Kokkos::deep_copy(dev, hs_a, 6);
+    ASSERT_TRUE(run_check(hs_a, 6));
+  }
+  {
+    Kokkos::deep_copy(dev, s_b, s_a);
+    ASSERT_TRUE(run_check(s_b, 5));
+  }
+  {
+    Kokkos::deep_copy(dev, hs_b, hs_a);
+    ASSERT_TRUE(run_check(hs_b, 6));
+  }
+  if (DevExecCanAccessHost || HostExecCanAccessDev) {
+    {
+      Kokkos::deep_copy(dev, hs_b, s_b);
+      ASSERT_TRUE(run_check(hs_b, 5));
+    }
+    {
+      Kokkos::deep_copy(dev, s_a, hs_a);
+      ASSERT_TRUE(run_check(s_a, 6));
+    }
+  }
 
-TEST(TEST_CATEGORY, view_copy_tests) {
-  // Only include this file to be compiled with CudaUVM and CudaHostPinned
-  TestViewCopy<TEST_EXECSPACE>::test_view_copy(4, 2, 3);
-  TestViewCopy<TEST_EXECSPACE>::test_view_copy(4, 2, 0);
+  // Contiguous copies
+  {
+    Kokkos::deep_copy(host, a, 2);
+    ASSERT_TRUE(run_check(a, 2));
+  }
+  {
+    Kokkos::deep_copy(host, b, 3);
+    ASSERT_TRUE(run_check(b, 3));
+  }
+  {
+    Kokkos::deep_copy(host, h_a, 4);
+    ASSERT_TRUE(run_check(h_a, 4));
+  }
+  {
+    Kokkos::deep_copy(host, a, b);
+    ASSERT_TRUE(run_check(a, 3));
+  }
+  {
+    Kokkos::deep_copy(host, h_b, h_a);
+    ASSERT_TRUE(run_check(h_b, 4));
+  }
+  {
+    Kokkos::deep_copy(host, h_a, a);
+    ASSERT_TRUE(run_check(h_a, 3));
+  }
+  {
+    Kokkos::deep_copy(host, b, h_b);
+    ASSERT_TRUE(run_check(b, 4));
+  }
+  // Non contiguous copes
+  {
+    Kokkos::deep_copy(host, s_a, 5);
+    ASSERT_TRUE(run_check(s_a, 5));
+  }
+  {
+    Kokkos::deep_copy(host, hs_a, 6);
+    ASSERT_TRUE(run_check(hs_a, 6));
+  }
+  {
+    Kokkos::deep_copy(host, s_b, s_a);
+    ASSERT_TRUE(run_check(s_b, 5));
+  }
+  {
+    Kokkos::deep_copy(host, hs_b, hs_a);
+    ASSERT_TRUE(run_check(hs_b, 6));
+  }
+  if (DevExecCanAccessHost || HostExecCanAccessDev) {
+    {
+      Kokkos::deep_copy(host, hs_b, s_b);
+      ASSERT_TRUE(run_check(hs_b, 5));
+    }
+    {
+      Kokkos::deep_copy(host, s_a, hs_a);
+      ASSERT_TRUE(run_check(s_a, 6));
+    }
+  }
 }
-#endif
 
 TEST(TEST_CATEGORY, view_copy_degenerated) {
   // Only include this file to be compiled with CudaUVM and CudaHostPinned
```

Comment:
I guess this comment is not true anymore if it ever was.

Reply:
fixed